File: data/transcoder_evaluation_gfg/c/HOW_TO_COMPUTE_MOD_OF_A_BIG_NUMBER.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

int f_gold ( char num [], int a ) {
  int res = 0;
  for ( int i = 0;
  i < strlen(num);
  i ++ ) res = ( res * 10 + ( int ) num [ i ] - '0' ) % a;
  return res;
}


int f_filled ( char num [], int a ) {}

int main(void) {
    int n_success = 0;
    char param0[][100] = {"RElCP","0139035510","00011110","TwanZWwLNXhFN","6247009752778","0100001011011","NCh","00714746542","101000100","MSTkXmlbPkV"};
    int param1[] = {13,44,86,66,55,33,75,54,93,78};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i],param1[i]) == f_gold(param0[i],param1[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/go/HOW_TO_COMPUTE_MOD_OF_A_BIG_NUMBER.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(num []byte, a int) int {
	var res int = 0
	for i := int(0); i < int(len(num)); i++ {
		res = (res*10 + int(num[i]) - int('0')) % a
	}
	return res
}
//TOFILL
func main() {
	var (
		n_success int           = 0
		param0    [10][]byte = [10][]byte{func() []byte {
			var t []byte
			copy(t[:], ([]byte)("RElCP"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("0139035510"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("00011110"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("TwanZWwLNXhFN"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("6247009752778"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("0100001011011"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("NCh"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("00714746542"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("101000100"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("MSTkXmlbPkV"))
			return t
		}()}
		param1 []int = []int{13, 44, 86, 66, 55, 33, 75, 54, 93, 78}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i][:], param1[i]) == f_gold(param0[i][:], param1[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}
